code/game/area/areas.dm

```diff
@@ -548,3 +555,131 @@ GLOBAL_LIST_EMPTY(teleportlocs)
+				if(SOUTHEAST_JUNCTION)
+					if(bitfield & SOUTH_JUNCTION || bitfield & EAST_JUNCTION)
+						continue
+					target_turf = locate(iterating_turf.x + 1, iterating_turf.y - 1, iterating_turf.z)
+				if(SOUTHWEST_JUNCTION)
+					if(bitfield & SOUTH_JUNCTION || bitfield & WEST_JUNCTION)
+						continue
+					target_turf = locate(iterating_turf.x - 1, iterating_turf.y - 1, iterating_turf.z)
+				if(NORTHWEST_JUNCTION)
+					if(bitfield & NORTH_JUNCTION || bitfield & WEST_JUNCTION)
+						continue
+					target_turf = locate(iterating_turf.x - 1, iterating_turf.y + 1, iterating_turf.z)
+			if(!target_turf)
+				continue
+			var/area/target_area = target_turf.loc
+			if(target_area == src)
+				continue
+			if(!target_area.outdoors || target_area.underground)
+				continue
+			bitfield ^= bit_step
+
+		if(!bitfield)
+			continue
+		adjacent_day_night_turf_cache[iterating_turf] = list(DAY_NIGHT_TURF_INDEX_BITFIELD, DAY_NIGHT_TURF_INDEX_APPEARANCE)
+		adjacent_day_night_turf_cache[iterating_turf][DAY_NIGHT_TURF_INDEX_BITFIELD] = bitfield
+		RegisterSignal(iterating_turf, COMSIG_PARENT_QDELETING, PROC_REF(clear_adjacent_turf))
+		if(iterating_turf.lighting_object)
+			iterating_turf.lighting_object.day_night_area = src
+
+	UNSETEMPTY(adjacent_day_night_turf_cache)
+
+/**
+ * Completely clears any adjacent turfs from the area while removing the effect.
+ */
+/area/proc/clear_adjacent_turfs()
+	for(var/turf/iterating_turf as anything in adjacent_day_night_turf_cache)
+		clear_adjacent_turf(iterating_turf)
+	adjacent_day_night_turf_cache = null
+
+/**
+ * Clears a signle turf from the adjacency turf cache.
+ * Arguments:
+ * * turf_to_clear - The turf we will unregister with and clear of any effects.
+ */
+/area/proc/clear_adjacent_turf(turf/turf_to_clear)
+	SIGNAL_HANDLER
+
+	turf_to_clear.underlays -= adjacent_day_night_turf_cache[turf_to_clear][DAY_NIGHT_TURF_INDEX_APPEARANCE]
+	adjacent_day_night_turf_cache -= turf_to_clear
+	UnregisterSignal(turf_to_clear, COMSIG_PARENT_QDELETING)
+
+/**
+ * A handler to set all adjacent turfs to the correct lighting.
+ */
+/area/proc/apply_day_night_turfs(datum/day_night_controller/incoming_controller)
+	SIGNAL_HANDLER
+
+	if(!incoming_controller)
+		return
+
+	for(var/turf/iterating_turf as anything in adjacent_day_night_turf_cache)
+		iterating_turf.underlays -= adjacent_day_night_turf_cache[iterating_turf][DAY_NIGHT_TURF_INDEX_APPEARANCE]
+		var/mutable_appearance/appearance_to_add = mutable_appearance(
+			icon = 'icons/effects/daynight_blend.dmi',
+			icon_state = "[adjacent_day_night_turf_cache[iterating_turf][DAY_NIGHT_TURF_INDEX_BITFIELD]]",
+			layer = DAY_NIGHT_LIGHTING_LAYER,
+			offset_spokesman = iterating_turf,
+			plane = LIGHTING_PLANE,
+			alpha = incoming_controller.current_light_alpha,
+			appearance_flags = RESET_COLOR | RESET_ALPHA | RESET_TRANSFORM
+		)
+		appearance_to_add.color = incoming_controller.current_light_color
```

Comment:
This feels like something that should be cached, with a color, alpha, bitfield and z level string key. Least going off gut it may not be quite such a big deal

Reply:
I feel like it's not really needed

Reply:
how hot is this code?